js/binance.js

```diff
@@ -3018,14 +3053,35 @@ module.exports = class binance extends Exchange {
         if ('maker' in trade) {
             takerOrMaker = trade['maker'] ? 'maker' : 'taker';
         }
+        if (('optionSide' in trade) || market['option']) {
```

Comment:
@Dan-krm next time, be careful when accessing `market` directly. As is we don't have any guarantees that the variable is not undefined so we need to add `safeMarket` or any other protection. I will add it this time nw